### ISSUE TITLE
commit.last_modified returns wrong date when retrieved from tags list

### ISSUE THREAD

**Issue by jemaf** [PyGithub/PyGithub#1064]
## The problem

Hey there, 

It seems that the value returned by the `last_modified` attribute of a given commit differs if we obtain this commit from the repository's tags list.

## Example

 ```{python}
from github import Github
gh = Github()
repo = gh.get_repo("PyGithub/PyGithub")
tag_commit = repo.get_tags()[0].commit
commit = repo.get_commit(tag_commit.sha)

# same commit sha
print("{} -- {}".format(tag_commit.sha, commit.sha))

# different last_modified dates
print("{} -- {}".format(tag_commit.last_modified, commit.last_modified))
```

## Environment

* Python: 3.7
* PyGithub: 1.43.5

**Comment by rgerkin** [comment]
This is related to #629 and I fixed this in #1032. It works for all practical purposes but some unrelated tests are failing so I will see if I can fix those.

**Comment by jakubLangr** [comment]
+1 for the problem. Thank you for all the great work! Was just about to report it. I'm looking forward to seeing this working from PyPI. 

On another note, if there's an easier way to figure out who last modified a repository _without_ making a separate request for commits, I'd also love to hear that.

**Comment by rgerkin** [comment]
The pypi problems are related to the process of retrieving the commits either leaving a file open somewhere or editing some sort of metadata, which then conflicts with the unit test's expectations.  I'm not sure it has any practical implication, but if someone is able to track this down (I wrote a more detailed note about it #1032) it would probably help get the PR accepted and this long-standing bug fixed.

**Comment by jakubLangr** [comment]
Hi, thanks for your quick reply, but I am not sure I understand it. So I thought this was already fixed in the repo and will be fixed for everyone once it gets deployed onto PyPI in a next release or something. 

Or any other suggestions on figuring out who did the last commit?

**Comment by rgerkin** [comment]
As far as I know, it is still a problem in master and the only fix is the one I submitted in #1032 that hasn't been accepted yet because of unrelated unit test failures.

**Comment by rgerkin** [comment]
This really is fixed in #1032, but no one has yet figured out why the unit tests are failing on Travis.

**Comment by rgerkin** [comment]
Fixed in #1032.
